dmodel/dm-query.c

```diff
@@ -577,51 +576,66 @@ get_title_clause (XapianQueryParser *qp,
                   GError **error_out)
 {
   g_autoptr(GError) error = NULL;
+  g_autoptr(XapianQuery) base_clause = NULL;
+  g_autoptr(XapianQuery) corrected_clause = NULL;
+  g_autoptr(XapianQuery) retval = NULL;
+
   guint terms_length = g_strv_length (terms);
   guint corrected_terms_length = corrected_terms != NULL
                                ? g_strv_length (corrected_terms)
                                : 0;
 
-  guint max_length = MAX (terms_length, corrected_terms_length);
+  g_auto(GStrv) parser_terms = g_new0 (gchar *, terms_length + 1);
```

Comment:
The code as it's written now looks like it might not need to duplicate `terms` and `corrected_terms`, since they only need to be passed to `g_strjoinv()`.

Reply:
Oh, right! I'll put back that deduplication, actually, and indeed only need to use g_new0 the one time (for the deduplicated corrected_terms strv).